networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/encodings/ssz/DefaultRpcPayloadEncoder.java

```diff
@@ -22,22 +22,28 @@
 import tech.pegasys.teku.networking.eth2.rpc.core.encodings.RpcPayloadEncoder;
 import tech.pegasys.teku.ssz.sos.SimpleOffsetSerializable;
 
-public class SimpleOffsetSszEncoder<T> implements RpcPayloadEncoder<T> {
+public class DefaultRpcPayloadEncoder<T> implements RpcPayloadEncoder<T> {
   private static final Logger LOG = LogManager.getLogger();
   private final Class<T> clazz;
 
-  public SimpleOffsetSszEncoder(final Class<T> clazz) {
+  public DefaultRpcPayloadEncoder(final Class<T> clazz) {
     this.clazz = clazz;
   }
 
   @Override
   public Bytes encode(final T message) {
-    return SimpleOffsetSerializer.serialize((SimpleOffsetSerializable) message);
+    return message instanceof Bytes
+        ? (Bytes) message
+        : SimpleOffsetSerializer.serialize((SimpleOffsetSerializable) message);
   }
 
+  @SuppressWarnings("unchecked")
   @Override
   public T decode(final Bytes message) throws RpcException {
     try {
+      if (clazz.equals(Bytes.class)) {
+        return (T) message;
```

Comment:
nit: shouldn't we be casting to (Bytes) here since the clazz is Bytes.class anyway? or does the compiler not allow that? or lastly is that just unnecessary?

Reply:
we're basically bypassing the decoder if its already bytes. the compiler wasn't happy! This is the other half of the encode bypass basically. If we don't bypass it here, then we attempt to go and decode it, and it fails.

Reply:
Oh yeah I understand that. I mean, on line 45, can we not do:
```suggestion
        return (Bytes) message;
```

Reply:
No because the compiler doesn't know that `T` is `Bytes`.  We do because we can make the leap from `Class<T> clazz` being equal to `Bytes.class` to knowing that `T` is `Bytes` but the compiler can't.